snippets/rp2350-ram-boot/rp2350-ram-boot.overlay

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2025 Dmitrii Sharshakov
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* We should partition SRAM between executable and data */
+/delete-node/ &sram0;
+
+&flash0 {
+    /* remove image definition and code partition in Flash */
+    /delete-node/ partitions;
+};
+
+#define RP2350_RAM_TOTAL_SIZE DT_SIZE_K(520)
+#define RP2350_RAM_CODE_SIZE  DT_SIZE_K(256)
+// Has to be adjusted as 0x20000000 + RP2350_RAM_CODE_SIZE if code partition is resized
+#define RP2350_RAM_OFFSET     20040000
```

Comment:
Is there a less ugly solution to plug a computed value in `memory@RP2350_RAM_OFFSET` unit address?

Reply:
Current code resize:
```diff
 #define RP2350_RAM_TOTAL_SIZE DT_SIZE_K(520)
-#define RP2350_RAM_CODE_SIZE  DT_SIZE_K(256)
+#define RP2350_RAM_CODE_SIZE  DT_SIZE_K(64)
 // Has to be adjusted as 0x20000000 + RP2350_RAM_CODE_SIZE if code partition is resized
-#define RP2350_RAM_OFFSET     20040000
+#define RP2350_RAM_OFFSET     20010000
 #define RP2350_IMAGE_DEF_SIZE 0x100
```